packages/server/package.json

```diff
@@ -12,14 +12,14 @@
     "devDropDb": "node dist/app.js --env dev --drop-db",
     "start": "node dist/app.js",
     "generateTypes": "rm -f db-buildTypes.sqlite && yarn run start --env buildTypes migrate latest && node dist/tools/generateTypes.js && mv db-buildTypes.sqlite schema.sqlite",
-    "tsc": "tsc --project tsconfig.json",
+    "tsc": "tsc --build",
```

Comment:
I added TypeScript support to the `public` directory on Joplin Server by creating a separate `tsconfig.json`.

This seems necessary because:
- The existing `tsconfig.json` outputs to a separate `dist/` folder (which does not contain the `public/` folder).
- The existing `tsconfig.json` creates CommonJS modules and includes references to `exports` (e.g. `Object.defineProperty(exports, "__esModule", { value: true });`) that won't work in a browser (without extra libraries/configuration).

This is done using [TypeScript project references](https://www.typescriptlang.org/docs/handbook/project-references.html). This allows the main TypeScript files and the `public/` TypeScript files to be built with a single `tsc` call.

However, this also [causes `.d.ts` files to be emitted](https://www.typescriptlang.org/docs/handbook/project-references.html#composite) ("declaration" must be enabled). This isn't necessary. As such, it may make sense to instead create a Gulp task that handles building the different TypeScript components.

Alternatively, these scripts could be bundled with Webpack. (We aren't using `import`s at present, so for now at least, this isn't necessary).